### ISSUE TITLE
`cilium encrypt status` fails when agents are down

### ISSUE THREAD

**Issue by pchaigno** [cilium/cilium#31897]
In a sysdump for a cluster where all Cilium agents are down:
```
$ cat cilium-encrypt-status.md 
Error: Cannot get daemon encryption status: Get "http://localhost/v1/healthz": dial unix /var/run/cilium/cilium.sock: connect: no such file or directory
> Error while running 'cilium encrypt status':  exit status 1
```
That shouldn't be happening: we should display what we can from Linux state and BPF maps. Only the state of IPsec/WireGuard enablement should require a connection to the agent.

**Comment by chaunceyjiang** [comment]
> Only the state of IPsec/WireGuard enablement should require a connection to the agent.

It seems that there is no way to detect whether IPsec/WireGuard is enabled in `cilium-dbg`. In the `cilium agent`, we can check `option.Config.EnableIPSec` or `option.Config.EnableWireguard` to see if IPsec/WireGuard is enabled, but in `cilium-dbg`, `option.Config.` is not initialized. @pchaigno

**Comment by pchaigno** [comment]
Yep, I don't think we can get that information if the agent is down. We should display `Cannot connect to agent` or some similar message in its place.

**Comment by Oppenheimer2** [comment]
Hi @pchaigno 👋

I'm new to Cilium development but I'm eager to jump in and contribute! This issue tagged as "good first issue" caught my eye, and I'd love to take it on. I already have Cilium set up in my local environment, so I'm ready to dive in.

One question I have is: How can I check the status of a Cilium agent? Since I'm not familiar with the Cilium codebase, I'd appreciate some pointers on how to do this. Could you please provide some guidance or resources to help me tackle this? Any help would be greatly appreciated!

Looking forward to contributing to Cilium! 🚀

**Comment by pchaigno** [comment]
Hi @Oppenheimer2!
> Since I'm not familiar with the Cilium codebase, I'd appreciate some pointers on how to do this. Could you please provide some guidance or resources to help me tackle this?

To find the code where `cilium encrypt status` is implemented, I'd suggest to run that command, take one of the output strings (ex. `Keys in use`) and grep that in the codebase. The same approach often works to identify the code related to open issues.

> How can I check the status of a Cilium agent?

Reading the above code and checking what happens when the agent is down should help identify how we can modify the code to test whether the agent is up or not.

**Comment by yogesh1801** [comment]
@pchaigno 
Can i work on this issue? i am starting a new contributor

**Comment by pchaigno** [comment]
Sure, anyone is welcome to work on this issue.

**Comment by parlakisik** [comment]
Is this issue still valid ? if it is , can you assign to me ? @pchaigno

**Comment by bersoare** [comment]
this one seems inactive, i'll take a look
